### ISSUE TITLE
Unvisible comments

### ISSUE THREAD

**Issue by tobbeanderberg** [agical/friendsunhosted#19]
When someone that you are not friends with makes comments in a conversation, a note about this should be visible in the conversation. e.g. "pelle@5apps.com made a comment". An add-link for that person could also be added next to the remark.

**Comment by brolund** [comment]
Yes. The idea is that anyone participating in a discussion also adds information about who else they see in the conversation. This is aggregated and presented like you describe. An option could be to just download their data as well, and if it is public like it is now, just present it without having to add a friend.

Later on, when we add privacy features, a user might opt to not have their names spread. The application should then honor that and add "+1 hidden" for those to make it easier to understand when something in a conversation is hidden.

**Comment by jancborchardt** [comment]
If anyone makes a comment in a conversation you can see, the comment should be displayed. It should depend on the visibility of the conversation parent, not of the comment itself. Otherwise it can be confusing if you don’t see all messages (and you might not want to add that person as friend).
Next to their name, as Tobbe said, should be an add-link for that person.

**Comment by brolund** [comment]
Things get more complicated once we start adding privacy, but I think we'll have to go in that direction at some point.

We have one idea of a thread-key that you use to protect a thread, and everyone adding to that thread uses that key. Exactly how to protect the key without adding insane amounts of keys, i don't know.

**Comment by johlrogge** [comment]
I think that public conversations should be seen as potentially completely public. So a thread has a symetrical key and it will propagate to friends of friends as friends comment in the thread.

If you want a conversation to be restricted, then it should be explicitly restricted, perhaps with invites. Basically, same as on facebook, if you comment in a friends thread their friends can read it.

In order not to have to fetch all friends of friends comments at once they can be hidden behind "more..." links and fetched on demand. And actively for threads that are open.
